src/diagnostics.rs

```diff
@@ -28,7 +28,7 @@ pub fn publish_diagnostics(params: Params, ctx: &mut Context) {
         .rev()
         .map(|x| {
             format!(
-                "{}|{}",
+                "'{}|{}'",
```

Comment:
Yes would be great to have a test because this is non-obvious (why would we need quoting if we always get something like `1.1,2.2|DiagnosticError` here?)

Something like

```sh
#!/bin/sh

# REQUIRES: command -v clangd

. test/lib.sh

cat >> .config/kak/kakrc << EOF
set-option global lsp_diagnostic_line_error_sign ' '
EOF

cat > main.c << EOF
syntax error
EOF

test_tmux_kak_start main.c
test_tmux capture-pane -p | grep syntax.error
# CHECK: {{ }}syntax error
# TODO add some other line-flags highlighter, so we can see the fix
```

Then run that file (`test/clangd-diagnostic-gutter.sh`) to see the output and
`test/run test/clangd-diagnostic-gutter.sh` to check the output.

The tests are a bit underdocumented but the nice thing is that you can add `test_tmux attach` and see what's going on

Reply:
Oops, because I made a stupid mistake and edited the wrong brackets.
I first made this fix in my local `.config/kak/plugins/kak-lsp`, and then quickly forked and applied the same change to submit a PR.
Apparently, I edited the wrong line though (changed 31 instead of 55).

I will add the unit test later this week!

Reply:
I added quotes to the correct line now and amended my commit.

Reply:
> I first made this fix in my local `.config/kak/plugins/kak-lsp`, and then quickly forked and applied the same change to submit a PR.

BTW I think you could have directly used that repo, you just need to set the remote to `git@github.com:user/repo` I usually use `hub fork`, which does that automatically I think.

Thanks for the test, looks really nice!